planner/core/find_best_task.go

```diff
@@ -1957,6 +1971,10 @@ func (ds *DataSource) convertToTableScan(prop *property.PhysicalProperty, candid
 		mppTask = ts.addPushedDownSelectionToMppTask(mppTask, ds.stats.ScaleByExpectCnt(prop.ExpectedCnt))
 		return mppTask, nil
 	}
+	if ts.StoreType == kv.TiFlash && mppApplicable && ds.SCtx().GetSessionVars().IsMPPAllowed() {
+		// don't generate cop task for TiFlash if mpp task is applicable
+		return invalidTask, nil
```

Comment:
this is why? two tasks can't co-exist together? or the cop task can always win the cost comparison

Reply:
Can two tasks can co-exist. But we always prefer MPP tasks if it is applicable. But our cost model may not be able to make the correct decision and choose cop task as the final plan, which is not what we want.